record/record.controller.js

```diff
@@ -116,52 +120,71 @@
         };
 
         /**
+         * {boolean} outputWVersion - true if version should be included in output
          * The following cases need to be handled for the resolverImplicitCatalog value:
-         *  - if undefined:                                 use current chaise path
+         *  - if null:                                      use current chaise path
          *  - if false:                                     /id/currCatalog/RID
+         *  - if undefined:                                 same as false
          *  - if resolverImplicitCatalog == currCatalog:    /id/RID
          *  - if resolverImplicitCatalog != currCatalog:    /id/resolverImplicitCatalog/RID
          **/
-        var resolvePermalink = function (tuple) {
+        var resolvePermalink = function (tuple, outputWVersion) {
             var resolverId = chaiseConfig.resolverImplicitCatalog;
             var currCatalog = $rootScope.reference.location.catalogId;
 
-            // null or no RID
-            if (resolverId === null || !tuple.data || !tuple.data.RID) {
-                return $window.location.href;
+            function rewriteUrlWithoutVersion() {
+                if (!outputWVersion && $rootScope.reference.location.version) {
+                    var hashNoVersion = $window.location.hash.replace($rootScope.reference.location.catalog, $rootScope.reference.location.catalogId);
+                    return $window.location.origin + $window.location.pathname + hashNoVersion;
+                } else {
+                    return $window.location.href;
+                }
             }
 
-            // if false we want the current catalog id
-            if (resolverId === false || resolverId === undefined) {
-                return $window.location.origin + "/id/" + currCatalog + "/" + tuple.data.RID;
+            // null or no RID
+            if (resolverId === null || !tuple.data || !tuple.data.RID) {
+                // only case that is different is when we have a version but don't want it in the output
+                return rewriteUrlWithoutVersion();
             }
 
             // if it's a number (isNaN tries to parse to integer before checking)
             if (!isNaN(resolverId)) {
-                var catalog = (resolverId != currCatalog) ? currCatalog+"/" : "";
-                return $window.location.origin + "/id/" + catalog + tuple.data.RID;
+                var catalog = (resolverId != currCatalog) ? currCatalog + "/" : "";
+                var output = $window.location.origin + "/id/" + catalog + tuple.data.RID;
+                return (outputWVersion) ? output + '@' + $rootScope.reference.location.version : output;
```

Comment:
This is what I meant. You're attaching the version of the current catalog to the output. The output might be using a different catalog id.